gameListFunctions.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+require_once 'gameClass.php';
+
+/**
+ * This code connects to the mySQL database
+ *
+ * @return PDO
+ */
+function dbConnection() : PDO{
+    $db = new PDO ('mysql:host=db;dbname=collection', 'root', 'password');
+
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    return $db;
+}
+
+/**
+ * This code gets an array of the full database
+ *
+ * @param PDO $db
+ * @return array
+ */
+function fullQuery(PDO $db) : Array
+{
+    $fullQuery = $db->prepare('SELECT `name`, `genre`, `length`, `price` FROM `games`;');
+    $fullQuery->execute();
+    return $fullQuery->fetchAll();
+}
+
+/**
+ * Generates the full list of objects based on the data
+ *
+ * @param array $data
+ * @return array
+ */
+function objectArray(Array $data) : Array{
+    $objectArray = [];
+    foreach($data as $game){
+        $gameObject = new Game($game['name'], $game['genre'], $game['length'], $game['price']);
+        $objectArray[] = $gameObject;
```

Comment:
This can be simplified slightly, there's no real need to store the game in $gmeObject first, you could just do this:

`$objectArray[] = new Game($game['name'], $game['genre'], $game['length'], $game['price']);`